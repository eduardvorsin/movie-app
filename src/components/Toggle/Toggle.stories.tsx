import type { Meta, StoryObj } from '@storybook/react';
import Toggle, { Props } from './Toggle';
import { ChangeEventHandler, useState } from 'react';
import { action } from '@storybook/addon-actions';
import MoonIcon from '../../assets/icons/moon.svg';
import SunIcon from '../../assets/icons/sun.svg';

type Sizes = Exclude<Props['size'], undefined>;
type Appearances = Exclude<Props['appearance'], undefined>;
const sizes: Sizes[] = ['regular', 'large'];
const appearances: Appearances[] = ['primary', 'secondary', 'warning', 'danger', 'success'];

type Variant = {
	appearance: Exclude<Props['appearance'], undefined>,
	label: string,
	labelHidden?: Exclude<Props['labelHidden'], undefined>,
	isDisabled?: Exclude<Props['isDisabled'], undefined>,
	hasIcons?: boolean,
}
const all: Record<Sizes, Variant[]> = {
	regular: [
		{
			appearance: 'primary',
			label: 'regular primary',
		},
		{
			appearance: 'secondary',
			label: 'regular secondary',
		},
		{
			appearance: 'success',
			label: 'regular warning',
		},
		{
			appearance: 'warning',
			label: 'regular danger',
		},
		{
			appearance: 'danger',
			label: 'regular danger',
		},
		{
			appearance: 'primary',
			label: 'regular disabled',
			isDisabled: true,
		},
		{
			appearance: 'primary',
			label: 'regular with icons',
			hasIcons: true,
		},
		{
			appearance: 'primary',
			label: 'regular label hidden',
			labelHidden: true,
		},
	],
	large: [
		{
			appearance: 'primary',
			label: 'large primary',
		},
		{
			appearance: 'secondary',
			label: 'large secondary',
		},
		{
			appearance: 'success',
			label: 'large warning',
		},
		{
			appearance: 'warning',
			label: 'large danger',
		},
		{
			appearance: 'danger',
			label: 'large danger',
		},
		{
			appearance: 'primary',
			label: 'large disabled',
			isDisabled: true,
		},
		{
			appearance: 'success',
			label: 'large with icons',
			hasIcons: true,
		},
		{
			appearance: 'secondary',
			label: 'large label hidden',
			labelHidden: true,
		},
	],
};

const meta: Meta<typeof Toggle> = {
	title: 'components/Toggle',
	tags: ['autodocs'],
	argTypes: {
		appearance: {
			description: 'Appearance for the Toggle component',
		},
		className: {
			description: 'Additional classes for the Toggle component'
		},
		id: {
			description: 'Id for input'
		},
		label: {
			description: 'Text description for the input field'
		},
		name: {
			description: 'Name for the input'
		},
		value: {
			description: 'Value for the input'
		},
		isChecked: {
			description: 'Is the Toggle component in a checked or unchecked state'
		},
		onChange: {
			control: false,
			description: 'A callback that is triggered when switching the state of the component'
		},
		size: {
			description: 'Dimensions of the Toggle component'
		},
		uncheckedIcon: {
			control: false,
			description: 'The icon that will be displayed on the left in the Toggle component'
		},
		checkedIcon: {
			control: false,
			description: 'The icon that will be displayed on the right in the Toggle component'
		},
		onBlur: {
			control: false,
			description: 'A callback that is triggered when the input loses focus'
		},
		onFocus: {
			control: false,
			description: 'A callback that is triggered when an input receives a focus'
		},
		labelHidden: {
			description: 'Allows you to leave the label visible only for screen readers'
		},
		isDisabled: {
			description: 'Allows you to make the Toggle component inactive for user interaction'
		},
		defaultChecked: {
			description: 'Initial state of the input'
		},
		testId: {
			description: 'Id for testing the component',
		}
	},
	component: Toggle,
	parameters: {
		docs: {
			description: {
				component: 'Toggle is an interface element designed to switch between two states, usually on and off. It is used to manage options, settings, or functionality of applications and web services.',
			},
		},
	},
}

export default meta;
type Story = StoryObj<typeof Toggle>;

const ToggleWithHooks = (props: Omit<Props, 'isChecked' | 'onChange'>) => {
	const [isChecked, setIsChecked] = useState<boolean>(props.defaultChecked ?? false);

	const changeHandler: ChangeEventHandler<HTMLInputElement> = (e) => {
		setIsChecked((prevState) => !prevState);
		action('Changed')(e);
	};

	return (
		<Toggle
			{...props}
			isChecked={isChecked}
			onChange={changeHandler}
		/>
	);
}
export const Default: Story = {
	args: {
		label: 'default',
		id: 'default',
	},
	parameters: {
		docs: {
			description: {
				story: 'The default state of the component',
			},
		},
	},
	render: (args) => (<ToggleWithHooks {...args} />),
};


export const Appearances: Story = {
	parameters: {
		docs: {
			description: {
				story: 'There are 5 values for the appearance property: primary, secondary, warning, success. Primary - for important actions. Secondary - for less important actions. Warning - to warn about potential risks. Danger - for reporting serious errors or irreversible actions. Success - to indicate successful actions',
			},
		},
	},
	render: () => (
		<div className='flex items-center gap-3 flex-wrap'>
			{appearances.map((appearance) => (
				<ToggleWithHooks
					key={appearance}
					id={appearance}
					label={appearance}
					value={appearance}
					name={appearance}
				/>
			))}
		</div>
	),
};


export const Sizes: Story = {
	parameters: {
		docs: {
			description: {
				story: 'Size options for the Toggle component. There are 2 types of them in total these are regular and large',
			},
		},
	},
	render: () => (
		<div className='flex items-center'>
			{sizes.map((size) => (
				<ToggleWithHooks
					className='mr-2 last:mr-0'
					key={size}
					id={size}
					label={size}
					value={size}
					size={size}
					name={size}
				/>
			))}
		</div>
	),
};

export const WithIcons: Story = {
	args: {
		label: 'withIcons',
		appearance: 'secondary',
		checkedIcon: <MoonIcon />,
		uncheckedIcon: <SunIcon />,
	},
	parameters: {
		docs: {
			description: {
				story: 'You can add icons inside the Toggle component to indicate the checked and unchecked state',
			},
		},
	},
	render: (args) => (<ToggleWithHooks {...args} />),
};

export const LabelHidden: Story = {
	args: {
		label: 'label hidden',
		appearance: 'secondary',
		labelHidden: true,
	},
	parameters: {
		docs: {
			description: {
				story: 'The label becomes visible only to screen readers',
			},
		},
	},
	render: (args) => (<ToggleWithHooks {...args} />),
};

export const Disabled: Story = {
	args: {
		label: 'disabled',
		appearance: 'primary',
		isDisabled: true,
	},
	parameters: {
		docs: {
			description: {
				story: 'Makes the Toggle component inactive for interaction. Useful when you need to temporarily restrict user actions to prevent accidental changes, for example when sending',
			},
		},
	},
	render: (args) => (<ToggleWithHooks {...args} />),
};

export const EventCallbacks: Story = {
	argTypes: {
		onFocus: {
			action: 'Focused',
		},
		onBlur: {
			action: 'Blured',
		},
	},
	args: {
		label: 'event callbacks',
		appearance: 'primary',
	},
	parameters: {
		docs: {
			description: {
				story: 'Callbacks are called when the state of the component changes, when the focus is received and lost',
			},
		},
	},
	render: (args) => (<ToggleWithHooks {...args} />),
};

export const All: Story = {
	parameters: {
		docs: {
			description: {
				story: '',
			},
		},
	},
	render: () => (
		<div className='flex items-start flex-wrap'>
			{(Object.keys(all) as Array<keyof typeof all>).map((size) => (
				<div
					key={size}
					className='flex flex-col items-end justify-start mr-6 last:mr-0'
				>
					{all[size].map((variant) => (
						<ToggleWithHooks
							className='mb-4 last:mb-0'
							key={`${size}-${variant.appearance}`}
							id={variant.label}
							name={variant.label}
							label={variant.label}
							value={variant.label}
							appearance={variant.appearance}
							size={size}
							isDisabled={variant.isDisabled}
							labelHidden={variant.labelHidden}
							checkedIcon={variant.hasIcons ? <MoonIcon></MoonIcon> : undefined}
							uncheckedIcon={variant.hasIcons ? <SunIcon></SunIcon> : undefined}
						/>
					))}
				</div>
			))}
		</div>
	),
};
