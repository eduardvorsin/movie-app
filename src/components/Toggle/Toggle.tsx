'use client';
import { GeneralProps } from '@/types/shared';
import { ChangeEventHandler, FocusEventHandler, ReactNode } from 'react';

export type Props = {
	id: string,
	label: string,
	name: string,
	value: string,
	isChecked: boolean,
	onChange: ChangeEventHandler<HTMLInputElement>,
	appearance?: 'primary' | 'secondary' | 'warning' | 'danger' | 'success',
	size?: 'regular' | 'large',
	uncheckedIcon?: ReactNode,
	checkedIcon?: ReactNode,
	onBlur?: FocusEventHandler<HTMLInputElement>,
	onFocus?: FocusEventHandler<HTMLInputElement>,
	labelHidden?: boolean,
	isDisabled?: boolean,
	defaultChecked?: boolean,
} & GeneralProps;

const appearances = {
	primary: 'bg-blue-700 dark:bg-blue-300',
	secondary: 'bg-neutral-800 dark:bg-dark-neutral-800',
	success: 'bg-green-700 dark:bg-green-300',
	danger: 'bg-red-700 dark:bg-red-300',
	warning: 'bg-yellow-700 dark:bg-yellow-300',
} as const;

export default function Toggle({
	id,
	label,
	name,
	value,
	size,
	uncheckedIcon,
	checkedIcon,
	appearance = 'primary',
	onChange,
	onBlur,
	onFocus,
	isChecked,
	labelHidden,
	isDisabled,
	defaultChecked,
	className,
	testId,
	...props
}: Props) {
	const inputWrapperClasses = [
		'text-neutral-100 dark:text-dark-neutral-100  relative transition-colors duration-150',
		appearances[appearance],
		size === 'large' ? 'w-[4.375rem] h-[2.1875rem] rounded-[2.1875rem]' : 'w-[3.75rem] h-[1.875rem] rounded-[1.875rem]',
		isDisabled ? 'opacity-disabled cursor-not-allowed' : '',
	].join(' ');

	const inputClasses = [
		'w-full h-full appearance-none cursor-pointer inline-flex p-1 relative disabled:cursor-not-allowed before:bg-neutral-100 dark:before:bg-dark-neutral-100 before:absolute before:rounded-[50%] before:z-100 before:transition-transform before:transition-colors before:duration-150 before:translate-x-0',
		size === 'large' ? 'rounded-[2.1875rem] before:w-[1.6875rem] before:h-[1.6875rem] checked:before:translate-x-[2.1875rem]' : 'rounded-[1.875rem] before:w-[1.375rem] before:h-[1.375rem] checked:before:translate-x-[1.875rem]',
	].join(' ');

	const labelClasses = [
		'text-neutral-800 dark:text-dark-neutral-800 mr-2 transition-colors duration-150',
		isDisabled ? 'opacity-disabled cursor-not-allowed' : '',
		labelHidden ? 'sr-only' : '',
	].join(' ');

	const iconWrapperClasses = [
		'pointer-events-none absolute top-1/2 -translate-y-1/2 z-0 fill-current',
		size === 'large' ? 'w-[calc(1.6875rem-2px)] h-[calc(1.6875rem-2px)]' : 'w-[calc(1.375rem-2px)] h-[calc(1.375rem-2px)]',
	].join(' ');

	return (
		<div
			className={`flex items-center ${className}`}
			data-testid={testId}
			{...props}
		>
			<label
				className={labelClasses}
				htmlFor={`${id}-switch-1`}
			>
				{label}
			</label>
			<div className={inputWrapperClasses}>
				<input
					className={inputClasses}
					role='switch'
					type='checkbox'
					onBlur={onBlur}
					onFocus={onFocus}
					onChange={onChange}
					defaultChecked={defaultChecked}
					checked={isChecked}
					disabled={isDisabled}
					name={name}
					value={value}
					id={`${id}-switch-1`}
				/>
				{uncheckedIcon && (
					<span
						className={`${iconWrapperClasses} left-[0.3125rem]`}
					>
						{uncheckedIcon}
					</span>
				)}
				{checkedIcon && (
					<span
						className={`${iconWrapperClasses} right-[0.3125rem]`}
					>
						{checkedIcon}
					</span>
				)}
			</div>
		</div>
	);
};