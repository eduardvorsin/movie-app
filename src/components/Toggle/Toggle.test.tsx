import { render, screen } from '@testing-library/react';
import Toggle from './Toggle';
import userEvent from '@testing-library/user-event';
import AddIcon from '../../assets/icons/add.svg';
import MinusIcon from '../../assets/icons/minus.svg';

describe('Toggle tests', () => {
	it('is rendered correctly', () => {
		const mockChangeHandler = jest.fn();
		render(<Toggle
			id='toggle'
			name='toggle'
			label='toggle'
			value='value'
			testId='toggle'
			isChecked={false}
			onChange={mockChangeHandler}
		/>);

		expect(screen.getByTestId<HTMLDivElement>('toggle')).toBeInTheDocument();
	});

	it('when clicking on the toggler, the mock function is called', async () => {
		const mockChangeHandler = jest.fn();
		const user = userEvent.setup();
		render(<Toggle
			id='toggle'
			name='toggle'
			label='toggle'
			value='value'
			testId='toggle'
			isChecked={false}
			onChange={mockChangeHandler}
		/>);

		await user.click(screen.getByRole<HTMLInputElement>('switch'));

		expect(mockChangeHandler).toHaveBeenCalledTimes(1);
	});

	it('when the focus hits the toggler, the mock function is called', async () => {
		const mockChangeHandler = jest.fn();
		const mockFocusHandler = jest.fn();
		const user = userEvent.setup();
		render(<Toggle
			id='toggle'
			name='toggle'
			label='toggle'
			value='value'
			testId='toggle'
			isChecked={false}
			onFocus={mockFocusHandler}
			onChange={mockChangeHandler}
		/>);

		await user.tab();

		expect(mockFocusHandler).toHaveBeenCalledTimes(1);
	});

	it('when the toggler loses focus, the mock function is called', async () => {
		const mockChangeHandler = jest.fn();
		const mockBlurHandler = jest.fn();
		const user = userEvent.setup();
		render(<Toggle
			id='toggle'
			name='toggle'
			label='toggle'
			value='value'
			testId='toggle'
			isChecked={false}
			onBlur={mockBlurHandler}
			onChange={mockChangeHandler}
		/>);

		await user.tab();
		await user.tab();

		expect(mockBlurHandler).toHaveBeenCalledTimes(1);
	});

	it('when isDisabled toggler becomes inactive', async () => {
		const mockChangeHandler = jest.fn();
		const user = userEvent.setup();
		render(<Toggle
			id='toggle'
			name='toggle'
			label='toggle'
			value='value'
			testId='toggle'
			isChecked={false}
			isDisabled
			onChange={mockChangeHandler}
		/>);

		await user.click(screen.getByRole<HTMLInputElement>('switch'));

		expect(mockChangeHandler).toHaveBeenCalledTimes(0);
	});

	it('if labelHidden is true then the label is shown only for screen readers', () => {
		const mockChangeHandler = jest.fn();
		render(<Toggle
			id='toggle'
			name='toggle'
			label='toggle'
			value='value'
			testId='toggle'
			isChecked={false}
			labelHidden
			onChange={mockChangeHandler}
		/>);

		expect(screen.getByText<HTMLLabelElement>('toggle')).toHaveClass('sr-only');
	});

	it('is a basic snapshot', () => {
		const mockChangeHandler = jest.fn();
		render(<Toggle
			id='toggle'
			name='toggle'
			label='toggle'
			value='value'
			testId='toggle'
			isChecked={false}
			onChange={mockChangeHandler}
		/>);

		expect(screen.getByTestId<HTMLDivElement>('toggle')).toMatchSnapshot();
	});

	it('snapshot with size equal to "regular"', () => {
		const mockChangeHandler = jest.fn();
		render(<Toggle
			id='toggle'
			size='regular'
			name='toggle'
			label='toggle'
			value='value'
			testId='toggle'
			isChecked={false}
			onChange={mockChangeHandler}
		/>);

		expect(screen.getByTestId<HTMLDivElement>('toggle')).toMatchSnapshot();
	});

	it('snapshot with size equal to "large"', () => {
		const mockChangeHandler = jest.fn();
		render(<Toggle
			id='toggle'
			size='large'
			name='toggle'
			label='toggle'
			value='value'
			testId='toggle'
			isChecked={false}
			onChange={mockChangeHandler}
		/>);

		expect(screen.getByTestId<HTMLDivElement>('toggle')).toMatchSnapshot();
	});

	it('snapshot with appearance equal to "primary"', () => {
		const mockChangeHandler = jest.fn();
		render(<Toggle
			id='toggle'
			appearance='primary'
			name='toggle'
			label='toggle'
			value='value'
			testId='toggle'
			isChecked={false}
			onChange={mockChangeHandler}
		/>);

		expect(screen.getByTestId<HTMLDivElement>('toggle')).toMatchSnapshot();
	});

	it('snapshot with appearance equal to "secondary"', () => {
		const mockChangeHandler = jest.fn();
		render(<Toggle
			id='toggle'
			appearance='secondary'
			name='toggle'
			label='toggle'
			value='value'
			testId='toggle'
			isChecked={false}
			onChange={mockChangeHandler}
		/>);

		expect(screen.getByTestId<HTMLDivElement>('toggle')).toMatchSnapshot();
	});

	it('snapshot with appearance equal to "warning"', () => {
		const mockChangeHandler = jest.fn();
		render(<Toggle
			id='toggle'
			appearance='warning'
			name='toggle'
			label='toggle'
			value='value'
			testId='toggle'
			isChecked={false}
			onChange={mockChangeHandler}
		/>);

		expect(screen.getByTestId<HTMLDivElement>('toggle')).toMatchSnapshot();
	});

	it('snapshot with appearance equal to "danger"', () => {
		const mockChangeHandler = jest.fn();
		render(<Toggle
			id='toggle'
			appearance='danger'
			name='toggle'
			label='toggle'
			value='value'
			testId='toggle'
			isChecked={false}
			onChange={mockChangeHandler}
		/>);

		expect(screen.getByTestId<HTMLDivElement>('toggle')).toMatchSnapshot();
	});

	it('snapshot with appearance equal to "success"', () => {
		const mockChangeHandler = jest.fn();
		render(<Toggle
			id='toggle'
			appearance='success'
			name='toggle'
			label='toggle'
			value='value'
			testId='toggle'
			isChecked={false}
			onChange={mockChangeHandler}
		/>);

		expect(screen.getByTestId<HTMLDivElement>('toggle')).toMatchSnapshot();
	});

	it('snapshot with labelHidden set to true', () => {
		const mockChangeHandler = jest.fn();
		render(<Toggle
			id='toggle'
			name='toggle'
			label='toggle'
			value='value'
			testId='toggle'
			isChecked={false}
			labelHidden
			onChange={mockChangeHandler}
		/>);

		expect(screen.getByTestId<HTMLDivElement>('toggle')).toMatchSnapshot();
	});

	it('snapshot with isDisabled set to true', () => {
		const mockChangeHandler = jest.fn();
		render(<Toggle
			id='toggle'
			name='toggle'
			label='toggle'
			value='value'
			testId='toggle'
			isChecked={false}
			isDisabled
			onChange={mockChangeHandler}
		/>);

		expect(screen.getByTestId<HTMLDivElement>('toggle')).toMatchSnapshot();
	});

	it('snapshot with icons', () => {
		const mockChangeHandler = jest.fn();
		render(<Toggle
			id='toggle'
			name='toggle'
			label='toggle'
			value='value'
			testId='toggle'
			isChecked={false}
			labelHidden
			onChange={mockChangeHandler}
			checkedIcon={<AddIcon />}
			uncheckedIcon={<MinusIcon />}
		/>);

		expect(screen.getByTestId<HTMLDivElement>('toggle')).toMatchSnapshot();
	});
});