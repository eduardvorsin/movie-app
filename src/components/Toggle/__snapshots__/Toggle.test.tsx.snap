// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle tests is a basic snapshot 1`] = `
<div
  class="flex items-center undefined"
  data-testid="toggle"
>
  <label
    class="text-neutral-800 dark:text-dark-neutral-800 mr-2  "
    for="toggle-switch-1"
  >
    toggle
  </label>
  <div
    class="text-neutral-100 dark:text-dark-neutral-100  relative bg-blue-700 dark:bg-blue-300 w-[3.75rem] h-[1.875rem] rounded-[1.875rem] "
  >
    <input
      class="w-full h-full appearance-none cursor-pointer inline-flex p-1 relative disabled:cursor-not-allowed before:bg-neutral-100 dark:before:bg-dark-neutral-100 before:absolute before:rounded-[50%] before:z-100 before:transition-transform before:duration-150 before:translate-x-0 rounded-[1.875rem] before:w-[1.375rem] before:h-[1.375rem] checked:before:translate-x-[1.875rem]"
      id="toggle-switch-1"
      name="toggle"
      role="switch"
      type="checkbox"
      value="value"
    />
  </div>
</div>
`;

exports[`Toggle tests snapshot with appearance equal to "danger" 1`] = `
<div
  class="flex items-center undefined"
  data-testid="toggle"
>
  <label
    class="text-neutral-800 dark:text-dark-neutral-800 mr-2  "
    for="toggle-switch-1"
  >
    toggle
  </label>
  <div
    class="text-neutral-100 dark:text-dark-neutral-100  relative bg-red-700 dark:bg-red-300 w-[3.75rem] h-[1.875rem] rounded-[1.875rem] "
  >
    <input
      class="w-full h-full appearance-none cursor-pointer inline-flex p-1 relative disabled:cursor-not-allowed before:bg-neutral-100 dark:before:bg-dark-neutral-100 before:absolute before:rounded-[50%] before:z-100 before:transition-transform before:duration-150 before:translate-x-0 rounded-[1.875rem] before:w-[1.375rem] before:h-[1.375rem] checked:before:translate-x-[1.875rem]"
      id="toggle-switch-1"
      name="toggle"
      role="switch"
      type="checkbox"
      value="value"
    />
  </div>
</div>
`;

exports[`Toggle tests snapshot with appearance equal to "primary" 1`] = `
<div
  class="flex items-center undefined"
  data-testid="toggle"
>
  <label
    class="text-neutral-800 dark:text-dark-neutral-800 mr-2  "
    for="toggle-switch-1"
  >
    toggle
  </label>
  <div
    class="text-neutral-100 dark:text-dark-neutral-100  relative bg-blue-700 dark:bg-blue-300 w-[3.75rem] h-[1.875rem] rounded-[1.875rem] "
  >
    <input
      class="w-full h-full appearance-none cursor-pointer inline-flex p-1 relative disabled:cursor-not-allowed before:bg-neutral-100 dark:before:bg-dark-neutral-100 before:absolute before:rounded-[50%] before:z-100 before:transition-transform before:duration-150 before:translate-x-0 rounded-[1.875rem] before:w-[1.375rem] before:h-[1.375rem] checked:before:translate-x-[1.875rem]"
      id="toggle-switch-1"
      name="toggle"
      role="switch"
      type="checkbox"
      value="value"
    />
  </div>
</div>
`;

exports[`Toggle tests snapshot with appearance equal to "secondary" 1`] = `
<div
  class="flex items-center undefined"
  data-testid="toggle"
>
  <label
    class="text-neutral-800 dark:text-dark-neutral-800 mr-2  "
    for="toggle-switch-1"
  >
    toggle
  </label>
  <div
    class="text-neutral-100 dark:text-dark-neutral-100  relative bg-neutral-800 dark:bg-dark-neutral-800 w-[3.75rem] h-[1.875rem] rounded-[1.875rem] "
  >
    <input
      class="w-full h-full appearance-none cursor-pointer inline-flex p-1 relative disabled:cursor-not-allowed before:bg-neutral-100 dark:before:bg-dark-neutral-100 before:absolute before:rounded-[50%] before:z-100 before:transition-transform before:duration-150 before:translate-x-0 rounded-[1.875rem] before:w-[1.375rem] before:h-[1.375rem] checked:before:translate-x-[1.875rem]"
      id="toggle-switch-1"
      name="toggle"
      role="switch"
      type="checkbox"
      value="value"
    />
  </div>
</div>
`;

exports[`Toggle tests snapshot with appearance equal to "success" 1`] = `
<div
  class="flex items-center undefined"
  data-testid="toggle"
>
  <label
    class="text-neutral-800 dark:text-dark-neutral-800 mr-2  "
    for="toggle-switch-1"
  >
    toggle
  </label>
  <div
    class="text-neutral-100 dark:text-dark-neutral-100  relative bg-green-700 dark:bg-green-300 w-[3.75rem] h-[1.875rem] rounded-[1.875rem] "
  >
    <input
      class="w-full h-full appearance-none cursor-pointer inline-flex p-1 relative disabled:cursor-not-allowed before:bg-neutral-100 dark:before:bg-dark-neutral-100 before:absolute before:rounded-[50%] before:z-100 before:transition-transform before:duration-150 before:translate-x-0 rounded-[1.875rem] before:w-[1.375rem] before:h-[1.375rem] checked:before:translate-x-[1.875rem]"
      id="toggle-switch-1"
      name="toggle"
      role="switch"
      type="checkbox"
      value="value"
    />
  </div>
</div>
`;

exports[`Toggle tests snapshot with appearance equal to "warning" 1`] = `
<div
  class="flex items-center undefined"
  data-testid="toggle"
>
  <label
    class="text-neutral-800 dark:text-dark-neutral-800 mr-2  "
    for="toggle-switch-1"
  >
    toggle
  </label>
  <div
    class="text-neutral-100 dark:text-dark-neutral-100  relative bg-yellow-700 dark:bg-yellow-300 w-[3.75rem] h-[1.875rem] rounded-[1.875rem] "
  >
    <input
      class="w-full h-full appearance-none cursor-pointer inline-flex p-1 relative disabled:cursor-not-allowed before:bg-neutral-100 dark:before:bg-dark-neutral-100 before:absolute before:rounded-[50%] before:z-100 before:transition-transform before:duration-150 before:translate-x-0 rounded-[1.875rem] before:w-[1.375rem] before:h-[1.375rem] checked:before:translate-x-[1.875rem]"
      id="toggle-switch-1"
      name="toggle"
      role="switch"
      type="checkbox"
      value="value"
    />
  </div>
</div>
`;

exports[`Toggle tests snapshot with icons 1`] = `
<div
  class="flex items-center undefined"
  data-testid="toggle"
>
  <label
    class="text-neutral-800 dark:text-dark-neutral-800 mr-2  sr-only"
    for="toggle-switch-1"
  >
    toggle
  </label>
  <div
    class="text-neutral-100 dark:text-dark-neutral-100  relative bg-blue-700 dark:bg-blue-300 w-[3.75rem] h-[1.875rem] rounded-[1.875rem] "
  >
    <input
      class="w-full h-full appearance-none cursor-pointer inline-flex p-1 relative disabled:cursor-not-allowed before:bg-neutral-100 dark:before:bg-dark-neutral-100 before:absolute before:rounded-[50%] before:z-100 before:transition-transform before:duration-150 before:translate-x-0 rounded-[1.875rem] before:w-[1.375rem] before:h-[1.375rem] checked:before:translate-x-[1.875rem]"
      id="toggle-switch-1"
      name="toggle"
      role="switch"
      type="checkbox"
      value="value"
    />
    <span
      class="pointer-events-none absolute top-1/2 -translate-y-1/2 z-0 fill-current w-[calc(1.375rem-2px)] h-[calc(1.375rem-2px)] left-[0.3125rem]"
    >
      <test-assets-stub />
    </span>
    <span
      class="pointer-events-none absolute top-1/2 -translate-y-1/2 z-0 fill-current w-[calc(1.375rem-2px)] h-[calc(1.375rem-2px)] right-[0.3125rem]"
    >
      <test-assets-stub />
    </span>
  </div>
</div>
`;

exports[`Toggle tests snapshot with isDisabled set to true 1`] = `
<div
  class="flex items-center undefined"
  data-testid="toggle"
>
  <label
    class="text-neutral-800 dark:text-dark-neutral-800 mr-2 opacity-disabled cursor-not-allowed "
    for="toggle-switch-1"
  >
    toggle
  </label>
  <div
    class="text-neutral-100 dark:text-dark-neutral-100  relative bg-blue-700 dark:bg-blue-300 w-[3.75rem] h-[1.875rem] rounded-[1.875rem] opacity-disabled cursor-not-allowed"
  >
    <input
      class="w-full h-full appearance-none cursor-pointer inline-flex p-1 relative disabled:cursor-not-allowed before:bg-neutral-100 dark:before:bg-dark-neutral-100 before:absolute before:rounded-[50%] before:z-100 before:transition-transform before:duration-150 before:translate-x-0 rounded-[1.875rem] before:w-[1.375rem] before:h-[1.375rem] checked:before:translate-x-[1.875rem]"
      disabled=""
      id="toggle-switch-1"
      name="toggle"
      role="switch"
      type="checkbox"
      value="value"
    />
  </div>
</div>
`;

exports[`Toggle tests snapshot with labelHidden set to true 1`] = `
<div
  class="flex items-center undefined"
  data-testid="toggle"
>
  <label
    class="text-neutral-800 dark:text-dark-neutral-800 mr-2  sr-only"
    for="toggle-switch-1"
  >
    toggle
  </label>
  <div
    class="text-neutral-100 dark:text-dark-neutral-100  relative bg-blue-700 dark:bg-blue-300 w-[3.75rem] h-[1.875rem] rounded-[1.875rem] "
  >
    <input
      class="w-full h-full appearance-none cursor-pointer inline-flex p-1 relative disabled:cursor-not-allowed before:bg-neutral-100 dark:before:bg-dark-neutral-100 before:absolute before:rounded-[50%] before:z-100 before:transition-transform before:duration-150 before:translate-x-0 rounded-[1.875rem] before:w-[1.375rem] before:h-[1.375rem] checked:before:translate-x-[1.875rem]"
      id="toggle-switch-1"
      name="toggle"
      role="switch"
      type="checkbox"
      value="value"
    />
  </div>
</div>
`;

exports[`Toggle tests snapshot with size equal to "large" 1`] = `
<div
  class="flex items-center undefined"
  data-testid="toggle"
>
  <label
    class="text-neutral-800 dark:text-dark-neutral-800 mr-2  "
    for="toggle-switch-1"
  >
    toggle
  </label>
  <div
    class="text-neutral-100 dark:text-dark-neutral-100  relative bg-blue-700 dark:bg-blue-300 w-[4.375rem] h-[2.1875rem] rounded-[2.1875rem] "
  >
    <input
      class="w-full h-full appearance-none cursor-pointer inline-flex p-1 relative disabled:cursor-not-allowed before:bg-neutral-100 dark:before:bg-dark-neutral-100 before:absolute before:rounded-[50%] before:z-100 before:transition-transform before:duration-150 before:translate-x-0 rounded-[2.1875rem] before:w-[1.6875rem] before:h-[1.6875rem] checked:before:translate-x-[2.1875rem]"
      id="toggle-switch-1"
      name="toggle"
      role="switch"
      type="checkbox"
      value="value"
    />
  </div>
</div>
`;

exports[`Toggle tests snapshot with size equal to "regular" 1`] = `
<div
  class="flex items-center undefined"
  data-testid="toggle"
>
  <label
    class="text-neutral-800 dark:text-dark-neutral-800 mr-2  "
    for="toggle-switch-1"
  >
    toggle
  </label>
  <div
    class="text-neutral-100 dark:text-dark-neutral-100  relative bg-blue-700 dark:bg-blue-300 w-[3.75rem] h-[1.875rem] rounded-[1.875rem] "
  >
    <input
      class="w-full h-full appearance-none cursor-pointer inline-flex p-1 relative disabled:cursor-not-allowed before:bg-neutral-100 dark:before:bg-dark-neutral-100 before:absolute before:rounded-[50%] before:z-100 before:transition-transform before:duration-150 before:translate-x-0 rounded-[1.875rem] before:w-[1.375rem] before:h-[1.375rem] checked:before:translate-x-[1.875rem]"
      id="toggle-switch-1"
      name="toggle"
      role="switch"
      type="checkbox"
      value="value"
    />
  </div>
</div>
`;
